### PR TITLE
Add allow-contract-caller for stacking pool (#592)

### LINKED ISSUE #592: [stacking-pool] Add allow-contract-caller
**Is your feature request related to a problem? Please describe.**
As a user, I want to be in control how I join the stacking pool. Therefore, I want to allow other contracts to call sign-pre-register and sign-register and similar methods on my behalf.

**Describe the solution you'd like**
Add `allow-contract-caller` and `revoke-contract-caller` and `get-contract-caller-allowance`

**Describe alternatives you've considered**
Remove any checks for `contract-caller` when calling `sign-pre-register` etc.

**Additional context**
This is a feature in pox-3 and other stacking related contracts like fast pool and pox-cycle-1-pool contracts.

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds allow-contract-caller function for stacking pool
* fixes #592 
* replaces space indentation with tab indentation
* adds allow-contract-caller in flow test
* increases code coverage for stacking pool to 48%

For easier review, hide white space diffs

### CODE REVIEW

**Review by FriendsFerdinand — APPROVED** [review]

File: sbtc-mini/contracts/sbtc-stacking-pool.clar [hunk 1]
```diff
@@ -307,150 +308,153 @@
...
+
+		;; Delegate-stx to their PoX address
+		(match (contract-call? .pox-3 delegate-stx amount-ustx (as-contract tx-sender) none (some pox-addr))
+			success true
+			error (asserts! false (err (+ (to-uint error) u60000))))
+
+		;; Delegate-stack-stx for next cycle
+		(match (as-contract (contract-call? .pox-3 delegate-stack-stx new-signer amount-ustx pox-addr burn-block-height u1))
+			success true
+			error (asserts! false (err (+ (to-uint error) u600000))))
```
Comment (MarvinJanssen) [inline sbtc-mini/contracts/sbtc-stacking-pool.clar:350]: Technically not part of this PR but we should make this a top level constant.

**Review by MarvinJanssen — APPROVED** [review]
Looks fine, we just have to be sure to communicate that an `until-burn-ht` of `none` will mean indefinite allowance.

File: sbtc-mini/contracts/sbtc-stacking-pool.clar (hunk 1 above)
Comment (friedger) [reply sbtc-mini/contracts/sbtc-stacking-pool.clar]: Updated
